profiles/permissions.py

```diff
@@ -20,3 +24,35 @@ def has_object_permission(self, request, view, profile):
             return True
 
         return profile.user == request.user
+
+
+class CanSeeIfNotPrivate(BasePermission):
+    """
+    Only owner of a profile can view and edit their profile.
+    Verified micromaster users can view other profiles to limited view only if
+    account_privacy of profile is set to public public_to_mm
+    """
+
+    def has_permission(self, request, view):
+        """
+        Implementation of the permission class.
+        """
+        profile = get_object_or_404(view.queryset, user__social_auth__uid=view.kwargs['user'])
```

Comment:
Can you replace `view.queryset` with `Profile`? It does not seem like `view.queryset` provides anything extra here and it makes it more difficult to mock